synapse/lib/webapp.py

```diff
@@ -102,14 +102,14 @@ def getFooByBar(self, bar):
         wapp.main()
 
     '''
-    def __init__(self, **settings):
+    def __init__(self, app_config={}, srv_config={}):
```

Comment:
can we pop app_config out of settings instead?
```
app_config = settings.pop('app_config', {})
```

otherwise this is an api breaking change.